analyzers/src/SonarAnalyzer.CFG/ShimLayer/CompilationOptionsWrapper.cs

```diff
@@ -0,0 +1,32 @@
+﻿// Copyright (c) Tunnel Vision Laboratories, LLC. All Rights Reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+namespace StyleCop.Analyzers.Lightup
+{
+    public readonly struct CompilationOptionsWrapper
+    {
+        internal const string WrappedTypeName = "Microsoft.CodeAnalysis.CompilationOptions";
+        private static readonly Type WrappedType;
+
+        private static readonly Func<CompilationOptions, object> SyntaxTreeOptionsProviderAccessor;
+        private readonly CompilationOptions node;
+
+        static CompilationOptionsWrapper()
+        {
+            WrappedType = WrapperHelper.GetWrappedType(typeof(CompilationOptionsWrapper));
+            SyntaxTreeOptionsProviderAccessor = LightupHelpers.CreateSyntaxPropertyAccessor<CompilationOptions, object>(WrappedType, nameof(SyntaxTreeOptionsProvider));
+        }
+
+        private CompilationOptionsWrapper(CompilationOptions node) =>
+            this.node = node;
+
+        public SyntaxTreeOptionsProviderWrapper SyntaxTreeOptionsProvider =>
+            node is null || WrappedType is null ? default : SyntaxTreeOptionsProviderWrapper.FromObject(SyntaxTreeOptionsProviderAccessor(node));
+
+        public static CompilationOptionsWrapper FromObject(CompilationOptions node) =>
+            node is null ? default : new(node);
```

Comment:
Other wrappers use `FromObject` method, which takes an `object` as a parameter. Would this also be useful here?

Reply:
I went for more strict contract, because we do have `CompilationOptions` in Roslyn 1.3.2 that we reference. Contrary to the other class that doesn't exist and `object` is our only option.
We should not need to construct it from `object`. And like this, I could have saved one type check.